src/main/scala/com/campudus/tableaux/database/domain/column.scala

```diff
@@ -64,105 +103,108 @@ sealed trait ColumnType[+A] extends DomainObject {
-  override val kind = DateType
-}
+case class CurrencyColumn(override val languageType: LanguageType)(override val columnInformation: ColumnInformation) extends SimpleValueColumn[Number](CurrencyType)(languageType)
 
-case class MultiDateTimeColumn(override val languageType: LanguageType)(override val table: Table, override val id: ColumnId, override val name: String, override val ordering: Ordering, override val identifier: Boolean, override val displayInfos: Seq[DisplayInfo], override val countryCodes: Option[Seq[String]]) extends MultiLanguageColumn[DateTime] {
-  override val kind = DateTimeType
-}
+case class BooleanColumn(override val languageType: LanguageType)(override val columnInformation: ColumnInformation) extends SimpleValueColumn[Boolean](BooleanType)(languageType)
+
+case class DateColumn(override val languageType: LanguageType)(override val columnInformation: ColumnInformation) extends SimpleValueColumn[LocalDate](DateType)(languageType)
+
+case class DateTimeColumn(override val languageType: LanguageType)(override val columnInformation: ColumnInformation) extends SimpleValueColumn[DateTime](DateTimeType)(languageType)
 
 /*
  * Special column types
  */
-case class LinkColumn[A](table: Table, id: ColumnId, to: ValueColumn[A], linkId: LinkId, linkDirection: LinkDirection, name: String, ordering: Ordering, override val identifier: Boolean, override val displayInfos: Seq[DisplayInfo]) extends ColumnType[Link[A]] {
+case class LinkColumn[A](override val columnInformation: ColumnInformation, to: ColumnType[A], linkId: LinkId, linkDirection: LinkDirection) extends ColumnType[Link[A]] {
   override val kind = LinkType
   override val languageType = to.languageType
 
   override def getJson: JsonObject = super.getJson mergeIn Json.obj("toTable" -> to.table.id, "toColumn" -> to.getJson)
 }
 
-case class AttachmentColumn(table: Table, id: ColumnId, name: String, ordering: Ordering, override val identifier: Boolean, override val displayInfos: Seq[DisplayInfo]) extends ColumnType[AttachmentFile] {
+case class AttachmentColumn(override val columnInformation: ColumnInformation) extends ColumnType[AttachmentFile] {
   override val kind = AttachmentType
+  override val languageType = LanguageNeutral()
 }
 
-case class ConcatColumn(table: Table, name: String, columns: Seq[ColumnType[_]]) extends ValueColumn[String] {
+case class ConcatColumn(override val columnInformation: ConcatColumnInformation, columns: Seq[ColumnType[_]]) extends ColumnType[String] {
   override val kind = ConcatType
 
   // ConcatColumn will be a multi-language
   // column in case any columns is also multi-language
-  override val languageType = !columns.exists(_.languageType match { case MultiLanguage | MultiCountry => true case _ => false }) match {
-    case true => MultiLanguage
-    case false => SingleLanguage
+  override val languageType = !columns.exists(_.languageType match { case _: MultiLanguage | _: MultiCountry => true case _ => false }) match {
```

Comment:
`ConcatColumn` can only be `LanguageNeutral`, `MultiLanguage` or `MultiCountry`. Problem `columns` can be all three.